clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java

```diff
@@ -201,7 +201,7 @@ public class ProducerConfig extends AbstractConfig {
     /** <code>buffer.memory</code> */
     public static final String BUFFER_MEMORY_CONFIG = "buffer.memory";
     private static final String BUFFER_MEMORY_DOC = "The total bytes of memory the producer can use to buffer records waiting to be sent to the server. If records are "
-                                                    + "sent faster than they can be delivered to the server the producer will block for <code>" + MAX_BLOCK_MS_CONFIG + "</code> after which it will throw an exception."
+                                                    + "sent faster than they can be delivered to the server the producer will block for <code>" + MAX_BLOCK_MS_CONFIG + "</code> after which it will end up with failed future."
```

Comment:
I'm not convinced by the description here. I know that some errors are reported as using futures, but I think that "fail with an exception" is probably better. If the interface in question uses a future, then that is reported as a failed future. If it's a callback, it's reported as a parameter on the callback. If it's a synchronous method, it's thrown. I recommend "fail with an exception" as a way of being accurate and also general at the same time.

Reply:
Makes sense. Let me revise